src/tests/index.test.js

```diff
@@ -0,0 +1,326 @@
+					evolv: {
+						javascript: {
+							variants: generateJsVariants(invokedJavascript)
+						}
+					}
+				};
+				global.document = new DocumentMock({elements: styleSheets.concat(scripts), styleSheets, scripts});
+				const client = new EvolvMock();
+				new EvolvAssetManager(client);
+				assert.equal(document.classList.classList.length, 0)
+			});
+
+			it('should not invoke javascript', () => {
+				const invokedJavascript = [];
+				global.window = {
+					location: {
+						href: 'https://test-site.com'
+					},
+					evolv: {
+						javascript: {
+							variants: generateJsVariants(invokedJavascript)
+						}
+					}
+				};
+				global.document = new DocumentMock({elements: styleSheets.concat(scripts), styleSheets, scripts});
+				const client = new EvolvMock();
+				new EvolvAssetManager(client);
+				assert.equal(invokedJavascript.length, 0)
+			});
+	
+			// it('should not confirm', () => {
+			// 	invokedJavascript = []
+			// 	global.document = new DocumentMock(elements);
+			// 	const client = new EvolvMock();
+			// 	main(client, options);
+			// 	assert.equal(client.confirmations, 0);
+			// });
+		});
+
+		describe('given active keys', () => {
+			const keys = ['web.page1', 'web.page1.variable1', 'web.page1.variable2'];
+
+			it('should add class names to the document classlist', () => {
+				const invokedJavascript = [];
+				global.window = {
+					location: {
+						href: 'https://test-site.com'
+					},
+					evolv: {
+						javascript: {
+							variants: generateJsVariants(invokedJavascript)
+						}
+					}
+				};
+				global.document = new DocumentMock({elements: styleSheets.concat(scripts), styleSheets, scripts});
+				const client = new EvolvMock(keys);
+				new EvolvAssetManager(client);
+				assert.equal(document.classList.classList.length, 3)
+				assert.deepEqual(document.classList.classList, [
+					'evolv_web_page1', 'evolv_web_page1_variable1', 'evolv_web_page1_variable2'])
+			});
+
+			it('should invoke javscript', () => {
+				const invokedJavascript = [];
+				global.window = {
+					location: {
+						href: 'https://test-site.com'
+					},
+					evolv: {
+						javascript: {
+							variants: generateJsVariants(invokedJavascript)
+						}
+					}
+				};
+				global.document = new DocumentMock({elements: styleSheets.concat(scripts), styleSheets, scripts});
+				const client = new EvolvMock(keys);
+				new EvolvAssetManager(client);
+				assert.equal(invokedJavascript.length, 3)
+				assert.ok(invokedJavascript.indexOf('evolv_web_page1') > -1)
+				assert.ok(invokedJavascript.indexOf('evolv_web_page1_variable1') > -1)
+				assert.ok(invokedJavascript.indexOf('evolv_web_page1_variable2') > -1)
+			});
+	
+			// it('should confirm once', () => {
+			// 	invokedJavascript = [];
+			// 	global.document = new DocumentMock(elements);
+			// 	const client = new EvolvMock(keys);
+			// 	main(client, options);
+			// 	setTimeout(assert.equal(client.confirmations, 1), 10000);
+			// });
+		});
+	});
+});
```

Comment:
Less so than the web-loader test, I do think an end-to-end test might be more appropriate here too. Many assertions depend on the behaviors of the mocks rather than the browser itself.

Reply:
Hmm i would argue that the mocks as scaffolding to make sure the asset manager is doing the correct thing in a black box environment. I think these tests are just a necessary as a end to end test.